src/compiler/checker.ts

```diff
@@ -6551,19 +6609,30 @@ module ts {
             }
         }
 
+        function createPromiseType(awaitedType: Type, func: FunctionLikeDeclaration): Type {
+            if (globalPromiseSymbol) {
+                awaitedType = getAwaitedType(awaitedType, awaitedType);
+                var promiseType = globalPromiseType || getDeclaredTypeOfSymbol(globalPromiseSymbol);

```

Comment:
If your language version is ES3 or ES5, globalPromiseSymbol is optional. I’ll try to use it if its defined globally, and if it isn’t I report an error.

From: Jason Freeman [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 3:53 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/checker.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24633116>:

> @@ -6551,19 +6609,30 @@ module ts {

>              }

>          }

>

> +        function createPromiseType(awaitedType: Type, func: FunctionLikeDeclaration): Type {

> +            if (globalPromiseSymbol) {

> +                awaitedType = getAwaitedType(awaitedType, awaitedType);

> +                var promiseType = globalPromiseType || getDeclaredTypeOfSymbol(globalPromiseSymbol);

Why would globalPromiseType be undefined, but globalPromiseSymbol be present?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24633116>.